### PR TITLE
refactor(console,phrases,core): reorg role details page

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

reorg role details page.
1. add `machine-to-machine roles` tab for m2m roles's details page to check what apps the role has been assigned to.
2. add `excludeRoleId` to GET /applications API (the same thing we did to GET /users API) to ignore m2m apps which have been assigned with the current m2m role (this is expected, you do not want to see all m2m apps when trying to assign new m2m apps for the current role).

## Testing

tested locally, will add UI tests later once the whole process is ready.
image
(can only assigned to 'm2m 002' app since this role has already been assigned to 'm2m 001' app)
image
image
image

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [ ] integration tests
- [ ] docs

OR

- [x] This PR is not applicable for the checklist

### CODE REVIEW

**Review by wangsijie — APPROVED** [review]

File: packages/console/src/pages/RoleDetails/RoleApplications/index.tsx [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+      setApplicationToBeDeleted(undefined);
+    } finally {
+      setIsDeleting(false);
+    }
+  };
+
+  return (
+    <>
+      <Table
+        className={styles.usersTable}
```
Comment (simeng-li) [inline packages/console/src/pages/RoleDetails/RoleApplications/index.tsx:86]: usersTable

Comment (simeng-li) [reply packages/console/src/pages/RoleDetails/RoleApplications/index.tsx:86]: If it takes extra effort to design a shared component, could define a shared style class first.

File: packages/console/src/pages/RoleDetails/RoleApplications/index.tsx [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+          },
+          {
+            title: null,
+            dataIndex: 'delete',
+            colSpan: 1,
+            render: (user) => (
+              <Tooltip content={t('general.remove')}>
+                <IconButton
+                  onClick={() => {
+                    setApplicationToBeDeleted(user);
```
Comment (simeng-li) [inline packages/console/src/pages/RoleDetails/RoleApplications/index.tsx:121]: user

File: packages/console/src/pages/RoleDetails/RoleApplications/index.tsx (hunk 1 above)
Comment (simeng-li) [reply packages/console/src/pages/RoleDetails/RoleApplications/index.tsx:86]: Better create a trackable issue to clean up these similar role assignment components.

File: packages/core/src/queries/application.ts [hunk 3]
```diff
@@ -48,21 +48,36 @@ const buildConditionArray = (conditions: SqlSqlToken[]) => {
...
-  const countApplications = async (search: Search) => {
+  const countApplications = async (search: Search, excludeApplicationIds: string[]) => {
     const { count } = await pool.one<{ count: string }>(sql`
       select count(*)
       from ${table}
-      ${buildConditionArray([buildApplicationConditions(search)])}
+      ${buildConditionArray([
+        excludeApplicationIds.length > 0
+          ? sql`${fields.id} not in (${sql.join(excludeApplicationIds, sql`, `)})`
+          : sql``,
```
Comment (simeng-li) [inline packages/core/src/queries/application.ts:58]: conditionalArraySql?

File: packages/core/src/queries/application.ts [hunk 4]
```diff
@@ -48,21 +48,36 @@ const buildConditionArray = (conditions: SqlSqlToken[]) => {
...
+    offset?: number
+  ) =>
     pool.any<Application>(sql`
       select ${sql.join(Object.values(fields), sql`, `)}
       from ${table}
-      ${buildConditionArray([buildApplicationConditions(search)])}
+      ${buildConditionArray([
+        excludeApplicationIds.length > 0
+          ? sql`${fields.id} not in (${sql.join(excludeApplicationIds, sql`, `)})`
+          : sql``,
```
Comment (simeng-li) [inline packages/core/src/queries/application.ts:78]: ditto

File: packages/console/src/pages/RoleDetails/RoleApplications/index.tsx (hunk 1 above)
Comment (darcyYe) [reply packages/console/src/pages/RoleDetails/RoleApplications/index.tsx:86]: This refactor has already been put on TODO list, see LOG-7102.

File: packages/core/src/queries/application.ts (hunk 3 above)
Comment (darcyYe) [reply packages/core/src/queries/application.ts:58]: the result of the trilogy operation is already part of the input of `conditionalArraySql`.

**Review by simeng-li — APPROVED** [review]
LGTM

**Pull request merged**